toolkit/tools/internal/resources/assets/efi/grub/grub.cfg

```diff
@@ -1,6 +1,9 @@
 search -n -u {{.BootUUID}} -s
-
 # If '/boot' is a seperate partition, BootUUID will point directly to '/boot'.
 # In this case we should omit the '/boot' prefix from all paths.
 set bootprefix={{.BootPrefix}}
+# For images using grub2-mkconfig, $prefix is the variable
+# We need to populate with the proper path to menuentry grub.cfg,
+# grubenv
+set prefix=($root)'/boot/grub2'
```

Comment:
why do we need this? grub detects the prefix and we really shouldn't be hardcoding it, right?

Reply:
In practice, on systems booting from an ESP partition, it thinks $prefix is /boot/efi/boot/grub2 which causes /boot/grub2/grubenv to not be sourced. 

This breaks things like grub2-editenv on Mariner.